api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -57,6 +57,19 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Bootstrap *string `json:"bootstrap,omitempty"`
+
+	// Config defines proxy level configuration options
+	//
+	// +optional
+	Config *ProxyConfig `json:"config,omitempty"`
```

Comment:
imo lets rm the top level `Config` field and directly add `Concurrency` here

Reply:
@arkodg actually, I'm fine either moving Concurrency to the top level or keeping it within the Config field as both satisfy my use case, but should we also take future extensibility into account to accommodate other proxy-level options when needed, wdyt?

Reply:
this field lives inside `EnvoyProxy.Spec` so we already have a envoy proxy specific home created, other fields at the same level are `logging` & `bootstrap` which are also envoy proxy cmdline args

Reply:
make sense, fixed.